version: '3.8'

services:
  update:
    build:
      dockerfile: Dockerfile.update
      context: .
    ports:
      - "8081:8080"
    networks:
      - mainnet_network
    environment:
      - DB_CONNECTION_STRING=${DB_CONNECTION_STRING_BACKUP}
      - NODE_CONNECTION_STRING=${NODE_CONNECTION_STRING}
      - ROUTER_ENDPOINT=${ROUTER_ENDPOINT}
      - RUST_LOG=info
    depends_on:
      db:
        condition: service_healthy

  fix:
    build:
      dockerfile: Dockerfile.fix
      context: .
    networks:
      - mainnet_network
    environment:
      - DB_CONNECTION_STRING=${DB_CONNECTION_STRING_BACKUP}
      - NODE_CONNECTION_STRING=${NODE_CONNECTION_STRING}
      - ROUTER_ENDPOINT=${ROUTER_ENDPOINT}
      - RUST_LOG=info
      - INTERVAL=${FIX_INTERVAL:-300}
    depends_on:
      db:
        condition: service_healthy

  db:
    image: postgres:16-alpine
    environment:
      - POSTGRES_USER=${POSTGRES_USER}
      - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
      - POSTGRES_DB=${POSTGRES_DB}
    ports:
      - "5432:5432"
    networks:
      - mainnet_network
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U ${POSTGRES_USER}"]
      interval: 5s
      timeout: 5s
      retries: 5

networks:
  mainnet_network:
volumes:
  db_mainnet_data: